src/transform/ReferenceCollection.js

```diff
@@ -0,0 +1,253 @@
+import ElementUtilities from './ElementUtilities'
+import NodeUtilities from './NodeUtilities'
+import Polyfill from './Polyfill'
+
+const REFERENCE_SELECTOR = '.reference'
+
+/**
+ * Is Citation.
+ * @param  {!string}  href
+ * @return {!boolean}
+ */
+const isCitation = href => href.indexOf('#cite_note') > -1
+
+/**
+ * Gets first child anchor.
+ * @param  {!HTMLElement} element
+ * @return {?HTMLAnchorElement}
+ */
+const getFirstChildAnchor = element => element.querySelector('A')
+
+/**
+ * Determines if node is a text node containing only whitespace.
+ * @param  {!Node}  node
+ * @return {!boolean}
+ */
+const isWhitespaceTextNode = node =>
+  !(!node || node.nodeType !== Node.TEXT_NODE || !node.textContent.match(/^\s+$/))
+
+/**
+ * Checks if element has a child anchor with a citation link.
+ * @param  {!HTMLElement}  element
+ * @return {!boolean}
+ */
+const hasCitationLink = element => {
+  try {
+    return isCitation(getFirstChildAnchor(element).getAttribute('href'))
+  } catch (e) {
+    return false
+  }
+}
+
+/**
+ * Get the reference text container.
+ * @param  {!Document} document
+ * @param  {!Node} sourceNode
+ * @return {?HTMLElement}
+ */
+const getRefTextContainer = (document, sourceNode) => {
+  const refTextContainerID = getFirstChildAnchor(sourceNode).getAttribute('href').slice(1)
+  const refTextContainer = document.getElementById(refTextContainerID)
+    || document.getElementById(decodeURIComponent(refTextContainerID))
```

Comment:
Do we really need to check both decoded and undecoded?

Reply:
Perhaps no longer, but I do recall some of these had funky encoding at one time...